test/parallel/test-cluster-shared-leak.js

```diff
@@ -15,14 +15,21 @@ if (cluster.isMaster) {
   worker1 = cluster.fork();
   worker1.on('message', common.mustCall(function() {
     worker2 = cluster.fork();
+    worker2.on('error', function(e) {
+      // EPIPE is OK on Windows
+      if (common.isWindows && e.code === 'EPIPE')
```

Comment:
I think I figured out why it's only afflicting `worker2` and how to rewrite the code without the special handling for Windows.

`worker2` gets `EPIPE` if we try to send to it before it's actually listening. This never happens with `worker1` because by the time we get to `worker1.send()`, we're in the `worker1` message handler, so it's obviously already listening. But there's no such protection for `worker2`. So the fix is to wait for the `listening` event to fire on `cluster` for `worker2` before doing all the `send()` stuff.  Running a stress test right now to confirm that's really the fix. Looks good so far. Will update this PR if that doesn't change. (Already confirmed that it still fails for Node 4.2.1, which is what we want.)